app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java

```diff
@@ -140,34 +160,70 @@ public void onSharedPreferenceChanged(SharedPreferences prefs, String key) {
         }
     };
 
+    public static byte getBridgeBattery() {
+        return bridgeBattery;
+    }
+
+
+    public boolean isUnitsMmol() {
+        return (PreferenceManager.getDefaultSharedPreferences(this).getString("units", "mmol").compareTo("mmol") != 0);
+    }
+
+    public static String getBridgeBatteryAsString() {
+        return String.format("%d", bridgeBattery);
+    }
+
     public void listenForChangeInSettings() {
         prefs.registerOnSharedPreferenceChangeListener(prefListener);
     }
 
     public void setRetryTimer() {
-        if (CollectionServiceStarter.isBTWixel(getApplicationContext())) {
-            long retry_in = (1000 * 60 * 2);
-            Log.d(TAG, "Restarting in: " + (retry_in / (60 * 1000)) + " minutes");
+        if (CollectionServiceStarter.isBTWixel(getApplicationContext()) || CollectionServiceStarter.isDexbridgeWixel(getApplicationContext())) {
+            long retry_in = (1000 * 62);
+            Log.d(TAG, "setRetryTimer: Restarting in: " + (retry_in / 1000) + " seconds");
             Calendar calendar = Calendar.getInstance();
             AlarmManager alarm = (AlarmManager) getSystemService(ALARM_SERVICE);
-            alarm.setExact(alarm.RTC_WAKEUP, calendar.getTimeInMillis() + retry_in, PendingIntent.getService(this, 0, new Intent(this, DexCollectionService.class), 0));
+            alarm.set(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + retry_in, PendingIntent.getService(this, 0, new Intent(this, DexCollectionService.class), 0));
         }
     }
 
     public void setFailoverTimer() { //Sometimes it gets stuck in limbo on 4.4, this should make it try again
-        if (CollectionServiceStarter.isBTWixel(getApplicationContext())) {
+        if (CollectionServiceStarter.isBTWixel(getApplicationContext()) || CollectionServiceStarter.isDexbridgeWixel(getApplicationContext())) {
             long retry_in = (1000 * 60 * 5);
-            Log.d(TAG, "Fallover Restarting in: " + (retry_in / (60 * 1000)) + " minutes");
+            Log.d(TAG, "setFailoverTimer: Fallover Restarting in: " + (retry_in / (60 * 1000)) + " minutes");
             Calendar calendar = Calendar.getInstance();
             AlarmManager alarm = (AlarmManager) getSystemService(ALARM_SERVICE);
-            alarm.set(alarm.RTC_WAKEUP, calendar.getTimeInMillis() + retry_in, PendingIntent.getService(this, 0, new Intent(this, DexCollectionService.class), 0));
+            alarm.set(AlarmManager.RTC_WAKEUP, calendar.getTimeInMillis() + retry_in, PendingIntent.getService(this, 0, new Intent(this, DexCollectionService.class), 0));
         } else {
             stopSelf();
         }
     }
 
-
-
+    /*    public void listenForChangeInSettings() {
```

Comment:
Not sure why it is commented out. Will investigate. Hasn't caused a problem in testing thus far. 

 
John Stevens
M - +61 418526349

> On 19 May 2015, at 23:58, Stephen Black <notifications@github.com> wrote:
> 
> In app/src/main/java/com/eveningoutpost/dexdrip/Services/DexCollectionService.java:
> 
> >          } else {
> >              stopSelf();
> >          }
> >      }
> >  
> > -
> > -
> > +    /*    public void listenForChangeInSettings() {
> ??? we need this
> 
> —
> Reply to this email directly or view it on GitHub.
>